src/args.rs

```diff
@@ -147,6 +111,83 @@ pub struct GenerateArgs {
     pub allow_commands: bool,
 }
 
+#[derive(Debug, Clone, Args)]
+pub struct TemplatePath {
+    /// Auto attempt to use as either `--git` or `--favorite`.
+    /// If either is specified explicitly, use as subfolder.
+    #[clap(required_unless_present_any(&["SpecificPath"]))]
+    pub auto_path: Option<String>,
+
+    /// Specifies a subfolder within the template repository to be used as the actual template.
+    #[clap()]
+    pub subfolder: Option<String>,
+
+    /// Git repository to clone template from. Can be a URL (like
+    /// `https://github.com/rust-cli/cli-template`), a path (relative or absolute), or an
+    /// `owner/repo` abbreviated GitHub URL (like `rust-cli/cli-template`).
+    ///
+    /// Note that cargo generate will first attempt to interpret the `owner/repo` form as a
+    /// relative path and only try a GitHub URL if the local path doesn't exist.
+    #[clap(short, long, group("SpecificPath"))]
+    pub git: Option<String>,
+
+    /// Branch to use when installing from git
+    #[clap(short, long)]
+    pub branch: Option<String>,
+
+    /// Local path to copy the template from. Can not be specified together with --git.
+    #[clap(short, long, group("SpecificPath"))]
+    pub path: Option<String>,
+
+    /// Generate a favorite template as defined in the config. In case the favorite is undefined,
+    /// use in place of the `--git` option, otherwise specifies the subfolder
+    #[clap(long, group("SpecificPath"))]
+    pub favorite: Option<String>,
+}
+
+impl TemplatePath {
+    /// # Panics
+    /// Will panic if no path to a template has been set at all,
+    /// which is never if Clap is initialized properly.
+    pub fn any_path(&self) -> String {
+        self.git
+            .as_ref()
+            .or(self.path.as_ref())
+            .or(self.favorite.as_ref())
+            .or(self.auto_path.as_ref())
+            .cloned()
+            .unwrap()
+    }
+
+    pub fn git(&self) -> Option<String> {
```

Comment:
nit: It would be enough, and more resource-friendly to expose here a `&str`. Let the caller decide if they need to take ownership of a clone.

Reply:
The same goes for the other accessor methods